hyperspace/cosmos/src/chain.rs

```diff
@@ -0,0 +1,107 @@
+use super::{error::Error, CosmosClient};
+use crate::finality_protocol::FinalityProtocol;
+use futures::Stream;
+use ibc_proto::google::protobuf::Any;
+use primitives::{Chain, IbcProvider};
+use prost::Message;
+use std::pin::Pin;
+use tendermint_rpc::Client;
+
+#[async_trait::async_trait]
+impl<H> Chain for CosmosClient<H>
+where
+	H: Clone + Send + Sync + 'static,
+{
+	fn name(&self) -> &str {
+		&*self.name
+	}
+
+	fn block_max_weight(&self) -> u64 {
+		todo!()
+	}
+
+	async fn estimate_weight(&self, messages: Vec<Any>) -> Result<u64, Self::Error> {
+		todo!()
+	}
+
+	async fn finality_notifications(
+		&self,
+	) -> Pin<Box<dyn Stream<Item = <Self as IbcProvider>::FinalityEvent> + Send + Sync>> {
+		match self.finality_protocol {
+			FinalityProtocol::Tendermint => {
+				todo!()
+			},
+		}
+	}
+
+	async fn submit(
+		&self,
+		messages: Vec<Any>,
+	) -> Result<(sp_core::H256, Option<sp_core::H256>), Error> {
+		
+		// Create SignerInfo by encoding the keybase
+		let mut pk_buf = Vec::new();
+		Message::encode(&self.keybase.public_key.to_pub().to_bytes(), &mut pk_buf)
+			.map_err(|e| Error::from(e.to_string()))?;
+
+		let pk_type = "/cosmos.crypto.secp256k1.PubKey".to_string();
+		let pk_any = Any { type_url: pk_type, value: pk_buf };
+		let single = Single { mode: 1 };
+		let sum_single = Some(Sum::Single(single));
+		let mode = Some(ModeInfo { sum: sum_single });
+		let signer_info =
+			SignerInfo { public_key: Some(pk_any), mode_info: mode, sequence: 0 as u64 };
```

Comment:
@Wizdave97 + @seunlanlege  can we start creating some `new()` methods for structs, in general, to avoid having to construct them inlined everywhere? I think it looks much better.

Reply:
In these particulars, I suspect be able to do so, since the structs are from raw proto types imported from `Ibc-proto` crate. Unless we define their peers' domain types with the `new()` method and impl `From` trait on them